expression/builtin_cast_vec.go

```
@@ -811,11 +811,57 @@ func (b *builtinCastStringAsTimeSig) vecEvalTime(input *chunk.Chunk, result *chu
 }
 
 func (b *builtinCastDecimalAsIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDecimalAsIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDecimal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDecimal(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	d64s := buf.Decimals()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+
+		// Round is needed for both unsigned and signed.
+		to := d64s[i]
+		err = d64s[i].Round(&to, 0, types.ModeHalfEven)
+		if err != nil {
```

It seems we should return the error here.